### ISSUE TITLE
Report number of executed artifact transforms in end-of-build summary on the console

### ISSUE THREAD

**Issue by lptr** [gradle/gradle#16744]
When we run a build, we report the number of actionable tasks on the console, like so:

```text
BUILD SUCCESSFUL in 1m 17s	
433 actionable tasks: 1 executed, 432 up-to-date
```

This emphasizes the false idea that the build is only made up of tasks. In reality, the execution phase of a build can involve a sizable portion of artifact transforms, too. This is especially true for Android builds, but the Kotlin plugin employs transforms outside of Android, too.

A few options to better represent the work done:

```text
BUILD SUCCESSFUL in 1m 17s	
433 actionable tasks: 1 executed, 432 up-to-date
2344 transforms: 7 executed, 2330 up-to-date, 7 loaded from cache
```

```text
BUILD SUCCESSFUL in 1m 17s	
2777 work units: 8 executed, 2762 up-to-date, 7 loaded from cache
```

Doing this would allow users to better understand what's happening during a build, and to comprehend how transform executions are being optimized. On a somewhat meta level mentioning transforms on the console would help people realize that transforms are present during a build.

---
cc: @gradle/execution

**Comment by lptr** [comment]
We could do this together with https://github.com/gradle/gradle/issues/16743.

**Comment by melix** [comment]
I'm not sure it makes a lot of sense for most users. Why talking about transforms and not dependencies downloaded, etc... ?

**Comment by lptr** [comment]
Yes, you're right that there are more kinds of work we could represent here. Or fewer. I think we should think about what we want to achieve with this summary in the first place. My argument is that what we do now (which is to single out tasks as the only kind of work we mention) is misleading and we should reconsider it.
